### DISCUSSION TITLE
Disable verify-profile required action for new realms in Keycloak >24.0.0

### DISCUSSION THREAD

**Discussion by SailReal** [keycloak/keycloak#29181] [Q&A]
Hey all, I have a question regarding https://github.com/keycloak/keycloak/issues/25985 (Keycloak >24.0.0 enables `verify-profile` required action by default)

We do not want to have `verify-profile` required action by default, how can we disable it in the `realm.json` file to import without adding all the other default `requiredActions`?

I'm aware that I can add

```
  "requiredActions": [
    {
      "alias": "VERIFY_PROFILE",
      "name": "Verify Profile",
      "providerId": "VERIFY_PROFILE",
      "enabled": false,
      "defaultAction": false,
      "priority": 10,
      "config": {}
    },
```

but then I would also have to add all the other `requiredActions` defaults, otherwise they are disabled, but these options can change over time, which we would have to keep track of, also it is possible that new ones are added in each Keycloak release.

Do you have a hint for me to only disable the `verify-profile` action? Maybe there is a special startup command or environment variable if we can not solve it in the `realm.json` file?

**Comment by codespearhead** [comment]
You can do so programatically after the server starts up through the [Keycloak Admin REST API](https://www.keycloak.org/docs-api/22.0.1/rest-api/index.html):

```bash
KEYCLOAK_BASE_URL="http://localhost:8080"
ADMIN_USERNAME="admin"
ADMIN_PASSWORD="admin"
REALM_NAME="master"
CLIENT_ID="admin-cli"

# 1. Make sure jq is installed. It'll help extract data from JSON responses.

jq --version

# 2. Get Master Token

MASTER_TOKEN_URL=$(curl --location --request GET \
  "$KEYCLOAK_BASE_URL/realms/$REALM_NAME/.well-known/openid-configuration" \
  | jq -r '.token_endpoint' \
)

MASTER_TOKEN=$(curl --location --request POST "$MASTER_TOKEN_URL" \
  --header 'Content-Type: application/x-www-form-urlencoded' \
  --data-urlencode "client_id=$CLIENT_ID" \
  --data-urlencode "username=$ADMIN_USERNAME" \
  --data-urlencode "password=$ADMIN_PASSWORD" \
  --data-urlencode 'grant_type=password' | \
  jq -r '.access_token' \
)

# 3. Disable the "Verify Profile" action:

curl -X PUT \
  "$KEYCLOAK_BASE_URL/admin/realms/$REALM_NAME/authentication/required-actions/VERIFY_PROFILE" \
  -H 'Content-Type: application/json' \
  -H "Authorization: Bearer $MASTER_TOKEN" \
  -d '{
      "alias": "VERIFY_PROFILE",
      "name": "Verify Profile",
      "providerId": "VERIFY_PROFILE",
      "enabled": true,
      "defaultAction": false,
      "priority": 10,
      "config": {}
  }'
```

However, I suggest you close this discussion and create a [feature request](https://github.com/keycloak/keycloak/issues/new?assignees=&labels=kind%2Ffeature%2Cstatus%2Ftriage&projects=&template=feature.yml) for the desired behavior in the `requiredActions` key.

**Reply by SailReal** [reply]
Thank you very much for your input. Unfortunately, this does not fit into our use case.

I wanted to see if I missed any options first, but if it really doesn't work straight away, I'd be happy to turn this discussion into a feature request.

**Reply by KapilBillore** [reply]
Hello , Did we make any changes to the disable verify profile option during startup  in keycloak 25?
